redstone/client.py

```diff
@@ -598,13 +615,190 @@ def unwrap(self, key_id, ciphertext, aad=None):
         resp = self._action(key_id, "unwrap", data)
         return base64.b64decode(resp["plaintext"].encode("utf-8"))
 
-    def rotate(self, key_id, payload=None):
+    def rewrap(self, key_id: str, ciphertext: str, aad: str = None):
+        """Use a root key to rewrap or reencrypt a data encryption key"""
+        # json body needs to be a UTF-8 string
+        if isinstance(ciphertext, bytes):
+            ciphertext = ciphertext.decode("utf-8")
+
+        data = {"ciphertext": ciphertext}
+
+        if aad:
+            data["aad"] = aad
+
+        return self._action(key_id, "rewrap", data)
+
+    def rotate(self, key_id: str, payload: bytes = None):
+        """Create a new version of a root key"""
         data = None
         if payload:
             data = {"payload": base64.b64encode(payload).decode("utf-8")}
 
         return self._action(key_id, "rotate", data)
 
+    def set_key_for_delete(self, key_id: str):
+        """Authorize deletion for a key with a dual authorization policy"""
+        resp = self.session.post("%s/api/v2/keys/%s/actions/setKeyForDeletion" % (self.endpoint_url, key_id))
+        self._validate_resp(resp)
+
+    def unset_key_for_delete(self, key_id: str):
+        """Remove an authorization for a key with a dual authorization policy"""
+        resp = self.session.post("%s/api/v2/keys/%s/actions/unsetKeyForDeletion" % (self.endpoint_url, key_id))
+        self._validate_resp(resp)
+
+    def disable_key(self, key_id: str):
+        """Disable operations for a key"""
+        resp = self.session.post("%s/api/v2/keys/%s/actions/disable" % (self.endpoint_url, key_id))
+        self._validate_resp(resp)
+
+    def enable_key(self, key_id: str):
+        """Enable operations for a key"""
+        resp = self.session.post("%s/api/v2/keys/%s/actions/enable" % (self.endpoint_url, key_id))
+        self._validate_resp(resp)
+
+    def get_registrations(self, key_id: str = None, crn: str = None):
+        """
+        Retrieve a list of registrations
+
+        If `key_id` is None (the default) all registrations for the instance are returned, otherwise
+        only the registrations associated with a specified root key are returned.
+        """
+        params = {}
+        if crn is not None:
+            params["urlEncodedResourceCRNQuery"] = crn
+
+        if key_id is not None:
+            url = "%s/api/v2/keys/%s/registrations" % (self.endpoint_url, key_id)
+        else:
+            url = "%s/api/v2/keys/registrations" % self.endpoint_url
+
+        resp = self.session.get(
+            url,
+            params=params
+        )
+
+        self._validate_resp(resp)
+        return resp.json()
+
+    def create_import_token(self, expiration: int = None, max_allowed_retrievals: int = None):
+        """Create an import token that can be used to import encrypted material as root keys."""
```

Comment:
Should document what unit or context expiration is. Is it a Unix timestamp, epoch seconds, or how many seconds from now it will expire in?